hail/src/main/scala/is/hail/expr/ir/functions/GenotypeFunctions.scala

```diff
@@ -39,7 +39,7 @@ object GenotypeFunctions extends RegistryFunctions {
         m2 - m)
     }
 
-    registerCode[Long]("dosage", TArray(tv("N", "float64")), TFloat64, null) { case (r, rt, (gpPType, gpOff)) =>
+    registerCode[Long]("dosage", TArray(tv("N", "float64")), TFloat64,  (pt: PType) => PFloat64()) { case (r, rt, (gpPType, gpOff)) =>
```

Comment:
this is totally unrelated to your change, but it looks like dosage and plDosage are incorrect -- they don't check for element missingness! Could you make a followup PR that fixes them? Just assigned a task to you in Asana for that.

I think throwing an exception on a missing element (like gqFromPL) is a perfectly fine solution.

Reply:
yep!